### ISSUE TITLE
Improve Pet AI

### ISSUE THREAD

**Issue by traaks** [traaks/echopet#13]
The current pet AI works pretty well for flat surfaces and surfaces that don't require jumping. With jumping I mean going up one block in case the ground is raised by one block. Pets do try to jump on top of a block when following their owner, but fail to actually get on top of it. They also don't seem to be very responsive in water, aren't able to swim in water, and aren't able to exit water without teleporting. There's also the old problem with ender dragons not moving.

In short, there are a ton of problems with the general pet AI and some individual ones. These can be fixed along the way and updates can be mentioned here. The point of this post is to simply declare the profuseness of the problem. It can be closed once problems with the AI are barely, if at all, noticeable.
